docs/components/select.md

```diff
@@ -367,6 +450,21 @@ export default {
     },
     deepClone(obj) {
       return JSON.parse(JSON.stringify(obj))
+    },
+    onKeywordChange (val) {
+      this.loading = true;
+      setTimeout(() => {
+        this.keyword = val;
+        this.itemsForAutosuggest = allItems.filter(item => item.label.includes(this.keyword)).map(item => Object.assign({}, item));
+        this.loading = false;
+      }, 400);
+    },
+    onFocus () {
+      this.loading = true;
+      setTimeout(() => {
+        this.itemsForAutosuggest = allItems.map(item => Object.assign({}, item));
+        this.loading = false;
+      }, 200);
```

Comment:
question: I'm not sure I understand the purpose of this `onFocus` handler. Can you explain?

Reply:
I think there might be cases where autosuggest should show options before any query is typed. For example, when creating a new SNI in KM, the autosuggest shows all existing certificates upon focus, before the end user types any keyword/query:
<img width="848" alt="image" src="https://user-images.githubusercontent.com/10095631/180928419-1b3a4eb9-f008-4a93-b80a-9016e08717cc.png">

This is what the `onFocus` handler does.

Reply:
Instead of the user having to manually add the `@focus` event handler, is there a way we could include this functionality by default in the `autosuggest` enabled `KSelect` component ?

Reply:
OK, let me think about it.

Reply:
I thought about this for a while but couldn't seem to find a better solution🤣.

We could have another prop, maybe something like `defaultSuggestions`, for users to pass in default options. But we already have `items`, and users can initialize their default options (maybe in the `created` hook) and pass them to `items`. This will have the same effect as adding a dedicated `defaultSuggestions` prop.

By the way, the example initializes default options in the `@focus` event handler instead of the `created` hook because I think this will eliminate unnecessary API requests. The end user may not click on the `KSelect` at all, so sending API requests in the `created` hook is a waste of resource. This also applies to if we have `defaultSuggestions`.

We could also turn `defaultSuggestions` into a method `getDefaultSuggestions`, and call this method inside `KSelect` itself when the `KInput` is focused. But this still requires users to pass in a method, which is not quite different from adding a `@focus` handler.

So do you have any better ideas?

Reply:
What about this: instead of attaching the `@focus` event to the KSelect, what if inside `KSelect.vue` `mounted` we emit the `query-change` event with an empty string as the value, which will trigger an initial load of the items?

```ts
mounted () {
  const inputElem = document.getElementById(this.selectInputId)
  if (inputElem) {
    this.inputWidth = inputElem.offsetWidth
    this.$emit('query-change', '')
  }
},
```

If I remove the `@focus` handler on the `KSelect` in the docs in the `### autosuggest` section, this seems to work. What do you think?

Reply:
Right, this could work. The downside of triggering `query-change` when `mounted` is that it still sends potentially unnecessary API request, if the end user does not interact with `KSelect` at all. That being said, I can implement this if it's acceptable.

Reply:
As an alternative solution, you could maybe do something like this:

1. Within `KSelect.vue` create a new `initialFocusTriggered: false` data variable
2. Detect the first focus of the field within the component; and on first focus, if `initialFocusTriggered` still equals false, trigger the event `emit` (which will load the items) and then set `initialFocusTriggered = true` to prevent it from firing again

Didn't test this, but I think it's possible

Reply:
But the event emitted with the first focus is also potentially unnecessary. If multiple `KSelect`s with autosuggest are present on a page, multiple API requests are sent once the page is loaded.

Reply:
Great! I'll implement this solution.

Reply:
Updated.

Reply:
I also updated the two examples for `autosuggest` so they look more like real life usages.